### PR TITLE
fix(components): [carousel] fix visible bug for async data

### LINKED ISSUE #5533: [Bug Report] 轮播图组件在设置了interval参数之后，异步获取数据用于渲染轮播图 ，组件会延迟interval相同的时间后才出现
### Element Plus version
1.3.0-beta.5

### OS/Browsers version
win10 / chrome 版本 97.0.4692.71（正式版本） （64 位）

### Vue version
3.2.25

### Reproduction Link
https://codesandbox.io/s/element-plus-spa-forked-8j74f?file=/src/App.vue

### Steps to reproduce
在onMounted中请求数据用于渲染轮播组件，组件会延迟与 interval 参数的时间之后才出现

### What is Expected?
组件立即出现

### What is actually happening?
轮播图组件在设置了interval参数之后，异步获取数据用于渲染轮播图 ，组件会延迟interval相同的时间后才出现

### PR DESCRIPTION
**Pull request by freedomlang**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

fix #5533: watch items to make sure `playSlides` would be excuted asap if there is data to show, so that the items can be visible

### CODE REVIEW

File: packages/components/carousel/src/main.vue [hunk 1]
```diff
@@ -355,6 +355,17 @@ export default defineComponent({
...
+    const stopWatchItems = watch(
+      () => items.value,
+      (val) => {
+        if (val.length) {
+          playSlides()
+          stopWatchItems()
+        }
+      },
+      { immediate: true, deep: true }
+    )
```
Comment (JeremyWuuuuu) [inline packages/components/carousel/src/main.vue:367]: **Question:** can we put them into `onMounted`? Can we write some tests for this change?

Comment (freedomlang) [reply packages/components/carousel/src/main.vue:367]: Yes, sure

File: packages/components/carousel/__tests__/carousel.spec.ts [hunk 2]
```diff
@@ -215,4 +215,26 @@ describe('Carousel', () => {
...
+      () => ({ carousels: [] })
+    )
+
+    await nextTick()
+    await wait(1000)
+    wrapper.setData({
+      carousels: [1, 2, 3],
+    })
+    await nextTick()
+    const items = wrapper.vm.$el.querySelectorAll('.el-carousel__item')
```
Comment (JeremyWuuuuu) [inline packages/components/carousel/__tests__/carousel.spec.ts:237]: **Nit** 

```ts
const items = wrapper.findAll('.el-carousel__item')
expect(items.at(0).classes('is-active')).toBe(true)
```

Comment (freedomlang) [reply packages/components/carousel/__tests__/carousel.spec.ts]: Look like Vue Test Utils 2 do not have `at` method for wrapper-array, i use `items[x]` instead

**Comment by xiaoxian521** [comment]
Hey, thank you very much for your contribution, but here are some conflicts that need to be solved by you. We can continue to review the code after solving. One thing to remind is that if the conflicts is still not resolved after `10` days, sorry, we will close this pr ❤️

**Comment by freedomlang** [comment]
> Hey, thank you very much for your contribution, but here are some conflicts that need to be solved by you. We can continue to review the code after solving. One thing to remind is that if the conflicts is still not resolved after `10` days, sorry, we will close this pr ❤️

Conflict resolved 👋

**Comment by freedomlang** [comment]
@xiaoxian521 Any update? 😄

File: packages/components/carousel/src/carousel.vue [hunk 3]
```diff
@@ -317,6 +317,19 @@ const resizeObserver = shallowRef<ReturnType<typeof useResizeObserver>>()
 onMounted(async () => {
   await nextTick()
 
+  if (items.value.length === 0) {
```
Comment (tolking) [inline packages/components/carousel/src/carousel.vue:320]: If the length of the initial items is not 0, then there is a problem.

[Demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGlkPVwiYXBwXCI+XG4gICAgPGVsLWNhcm91c2VsXG4gICAgICA6aW50ZXJ2YWw9XCIxMDAwMFwiXG4gICAgICBpbmRpY2F0b3ItcG9zaXRpb249XCJub25lXCJcbiAgICAgIHR5cGU9XCJjYXJkXCJcbiAgICAgIGhlaWdodD1cIjMwMHB4XCJcbiAgICAgIHdpZHRoPVwiMzAwcHhcIlxuICAgID5cbiAgICAgIDxlbC1jYXJvdXNlbC1pdGVtIHYtZm9yPVwiKGl0ZW0sIGluZGV4KSBpbiBkYXRhTGlzdFwiIDprZXk9XCJpbmRleFwiPlxuICAgICAgICB7eyBpdGVtIH19XG4gICAgICAgIDxlbC1pbWFnZVxuICAgICAgICAgIHNyYz1cImh0dHBzOi8vdmlhLnBsYWNlaG9sZGVyLmNvbS8zMDAucG5nLzA5Zi9mZmZcIlxuICAgICAgICAgIHN0eWxlPVwid2lkdGg6IDEwMCU7IGhlaWdodDogMTAwJVwiXG4gICAgICAgIC8+XG4gICAgICA8L2VsLWNhcm91c2VsLWl0ZW0+XG4gICAgPC9lbC1jYXJvdXNlbD5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0PlxuaW1wb3J0IHsgcmVmLCBvbk1vdW50ZWQgfSBmcm9tIFwidnVlXCI7XG5cbmV4cG9ydCBkZWZhdWx0IHtcbiAgc2V0dXAoKSB7XG4gICAgY29uc3QgZGF0YUxpc3QgPSByZWYoWzBdKTtcbiAgICBvbk1vdW50ZWQoKCkgPT4ge1xuICAgICAgLy8g5qih5ouf5pWw5o2u6K+35rGCXG4gICAgICBzZXRUaW1lb3V0KCgpID0+IHtcbiAgICAgICAgZGF0YUxpc3QudmFsdWUgPSBbMSwgMiwgM107XG4gICAgICAgIGNvbnNvbGUubG9nKFwi5pWw5o2uIOWKoOi9veWujOaIkFwiKTtcbiAgICAgIH0sIDEwMDApO1xuICAgICAgc2V0VGltZW91dCgoKSA9PiB7XG4gICAgICAgIGNvbnNvbGUubG9nKFwiMTDnp5Lov4fljrvkuoZcIik7XG4gICAgICB9LCAxMDAwMCk7XG4gICAgfSk7XG4gICAgLy8g5pq06Zyy57uZIHRlbXBsYXRlXG4gICAgcmV0dXJuIHtcbiAgICAgIGRhdGFMaXN0LFxuICAgIH07XG4gIH0sXG59O1xuPC9zY3JpcHQ+XG5cblxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTY1NDgtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiXG4gIH1cbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL3VucGtnLmNvbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vdW5wa2cuY29tL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTY1NDgtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL3VucGtnLmNvbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTY1NDgtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gIH0pXG59IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctNjU0OC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==)

**Comment by tolking** [comment]
similar PR #8904

**Comment by freedomlang** [comment]
> similar PR #8904

@tolking @holazz
Thanks for the review. Btw, since there is similar PR, is there any necessity for this pr? If not, i'll close it. 😂

**Pull request closed without merging**